src/main/java/org/jboss/logmanager/formatters/Formatters.java

```diff
@@ -471,7 +471,13 @@ public void renderRaw(final StringBuilder builder, final ExtLogRecord record) {
      * @return the format step
      */
     public static FormatStep hostnameFormatStep(final boolean leftJustify, final int minimumWidth, final boolean truncateBeginning, final int maximumWidth, final boolean qualified) {
-        return hostnameFormatStep(leftJustify, minimumWidth, truncateBeginning, maximumWidth, qualified ? null : "1");
+        return new SegmentedFormatStep(leftJustify, minimumWidth, truncateBeginning, maximumWidth, null) {
+            public String getSegmentedSubject(final ExtLogRecord record) {
+                final String hostName = record.getHostName();
+                final int idx = hostName.indexOf('.');
+                return idx == -1 ? hostName :hostName.substring(0, idx);
```

Comment:
The best fix would be for the segment count to start from the left end instead of the right end though.

Reply:
How about this one?

Reply:
Yeah. That's the precision I was going to use in this case. This would work, though the `qualified` is still ignored. Line 477 could just be `final int idx = qualified ? -1 : hostName.indxedOf('.');` though. We could fix the other one later.

Reply:
Something like this would probably work.
```
    public static FormatStep hostnameFormatStep(final boolean leftJustify, final int minimumWidth, final boolean truncateBeginning, final int maximumWidth, final String precision) {
        return new SegmentedFormatStep(leftJustify, minimumWidth, truncateBeginning, maximumWidth, null) {
            public String getSegmentedSubject(final ExtLogRecord record) {
                final String hostName = record.getHostName();
                // Check for dots
                if (PRECISION_INT_PATTERN.matcher(precision).matches()) {
                    int count = Integer.parseInt(precision);
                    int end = 0;
                    for (int i = 0; i < hostName.length(); i++) {
                        if (hostName.charAt(i) == '.') {
                            count--;
                            end = i;
                        }
                        if (count == 0) {
                            break;
                        }
                    }
                    if (end == 0 || count != 0) {
                        return hostName;
                    }
                    return hostName.substring(0, end);
                }
                return hostName;
            }
        };
    }
```

Reply:
I'm going to add the above for the precision and add some tests, then I'll merge this. I'll look at back porting to 2.0 as well.

Reply:
Actually. What might make sense is to use `%H{-1}` for right truncation and `%H{1}` for left truncation to keep it with what we have now.